codebase/src/components/edge.ts

```diff
@@ -1,127 +1,173 @@
-  // Make this not hard coded
-  //const canvas: HTMLCanvasElement = document.getElementById("GraphArea") as HTMLCanvasElement;
-  //const ctx = canvas.getContext("2d");
-  // Maybe optimize this using adjacency list later
-  if (ctx) {
-    for (let i = 0; i < edges.length; i++) {
-      let to: string = edges[i].to;
-      let from: string = edges[i].from;
-
-      //Self Edge
-      if (to == from) {
-        let nodeCords: number[] = [];
-        for (let x = 0; x < nodes.length; x++) {
-          let cur = nodes[x];
-          if (cur.id == to) {
-            nodeCords = [cur.x, cur.y];
-          }
-        }
-        //Maybe make angle opposite to the average of all other angles for that node
-        drawSelfEdge(ctx, (Math.PI), nodeCords[0], nodeCords[1], CIRCLER, edges[i]);
-        //drawSelfEdge(ctx, (Math.PI / 2), nodeCords[0], nodeCords[1], CIRCLER, edges[i]);
-        continue;
-      }
-
-      let toX = -1;
-      let toY = -1;
-      let fromX = -1;
-      let fromY = -1;
-      for (let x = 0; x < nodes.length; x++) {
-        let cur = nodes[x];
-        if (cur.id == to) {
-          toX = cur.x;
-          toY = cur.y;
-        }
-        else if (cur.id == from) {
-          fromX = cur.x;
-          fromY = cur.y;
-        }
-      }
-      if (toX == -1 || toY == -1 || fromX == -1 || fromY == -1) {
-        alert("One of these nodes does not exist");
-        return;
-      }
-      //drawDirectedEdge(ctx, fromX, fromY, toX, toY, edges[i], CIRCLER);
-      drawUndirectedEdge(ctx, fromX, fromY, toX, toY, edges[i], CIRCLER);
-    }
-  }
+	// Make this not hard coded
+	//const canvas: HTMLCanvasElement = document.getElementById("GraphArea") as HTMLCanvasElement;
+	//const ctx = canvas.getContext("2d");
+	//
+	// Maybe optimize this using adjacency list later
+	// TODO: Add support for editable text boxes
+	// TODO: Add bi-directional edges
+	// TODO: Add deletion of edges
+	if (ctx) {
+		for (let i = 0; i < edges.length; i++) {
+			let to: number = edges[i].to;
+			let from: number = edges[i].from;
+
+			//Self Edge
+			if (to == from) {
+				let nodeCords: number[] = [];
+				for (let x = 0; x < nodes.length; x++) {
+					let cur = nodes[x];
+					if (cur.id == to) {
+						nodeCords = [cur.x, cur.y];
+					}
+				}
+				//Maybe make angle opposite to the average of all other angles for that node
+				drawSelfEdge(ctx, (Math.PI), nodeCords[0], nodeCords[1], CIRCLER, edges[i]);
+				//drawSelfEdge(ctx, (Math.PI / 2), nodeCords[0], nodeCords[1], CIRCLER, edges[i]);
+				continue;
+			}
+
+			let toX = -1;
+			let toY = -1;
+			let fromX = -1;
+			let fromY = -1;
+			for (let x = 0; x < nodes.length; x++) {
+				let cur = nodes[x];
+				if (cur.id == to) {
+					toX = cur.x;
+					toY = cur.y;
+				}
+				else if (cur.id == from) {
+					fromX = cur.x;
+					fromY = cur.y;
+				}
+			}
+			if (toX == -1 || toY == -1 || fromX == -1 || fromY == -1) {
+				alert("One of these nodes does not exist");
```

Comment:
alert is user facing and not intended for error handling in JS/TS. you want to replace your return with `throw` then create a new instance of `Error` by passing it this message you have in alert